Patch by Zarafa <http://www.zarafa.com/> - when an email contains the same boundary to announce a new body
part directly one after another, vmime would have crashed because the body part added to the object with
size of (size_type)(-1). Thus having a body part of 4294967294 bytes on 32bit and 18446744073709551615
bytes on 64 bit systems.

--- libvmime-0.7.1/src/body.cpp						2005-03-18 22:26:46.000000000 +0100
+++ libvmime-0.7.1/src/body.cpp.double-empty-boundary			2007-11-01 16:45:20.688580500 +0100
@@ -173,7 +173,7 @@
 				++pos;
 			}
 
-			if (index > 0)
+			if (index > 0 && partStart < partEnd)
 			{
 				bodyPart* part = new bodyPart;
 
